qutebrowser/completion/models/sortfilter.py

```diff
@@ -140,21 +133,33 @@ def filterAcceptsRow(self, row, parent):
         if parent == QModelIndex() or not self.pattern:
             return True
 
-        try:
-            return self.srcmodel.custom_filter(self.pattern, row, parent)
-        except NotImplementedError:
-            for col in self.srcmodel.columns_to_filter:
-                idx = self.srcmodel.index(row, col, parent)
-                if not idx.isValid():
-                    # No entries in parent model
-                    continue
-                data = self.srcmodel.data(idx)
-                if not data:
-                    continue
-                elif self.pattern_re.search(data):
-                    return True
+        # first check the cache
```

Comment:
Funnily enough I *can* reproduce it with `--temp-basedir` (and only two rows):

```
09:52:06 DEBUG    completion completer:_get_new_completion:233 completions: *url*
09:52:06 DEBUG    completion debug:__exit__:256 Setting filter pattern took 0.000184 seconds.
09:52:06 DEBUG    completion completer:update_completion:348 New completion for ['open', 't']: UrlCompletionModel, with pattern 't'
09:52:06 DEBUG    completion sortfilter:filterAcceptsRow:150 ['https://duckduckgo.com/', 'DuckDuckGo']
09:52:06 DEBUG    completion sortfilter:filterAcceptsRow:150 ['https://duckduckgo.com/?q=TEST&t=hd', 'TEST at DuckDuckGo']
```

I did a quick look at the [C++ sourcecode](http://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/itemmodels/qsortfilterproxymodel.cpp?h=5.7.0) yesterday - I'm not sure what exactly `invalidate()` [does](https://github.com/qt/qtbase/blob/5.7.0/src/corelib/itemmodels/qsortfilterproxymodel.cpp#L2578-L2584) the refiltering, but at least calling `invalidateFilter()` [seems to](https://github.com/qt/qtbase/blob/5.7.0/src/corelib/itemmodels/qsortfilterproxymodel.cpp#L2607-L2611) immediately start a refiltering (via [`filter_changed`](https://github.com/qt/qtbase/blob/5.7.0/src/corelib/itemmodels/qsortfilterproxymodel.cpp#L1075-L1111) -> [`handle_filter_changed`](https://github.com/qt/qtbase/blob/5.7.0/src/corelib/itemmodels/qsortfilterproxymodel.cpp#L1117-L1156) -> `filterAcceptsRow`)